commitment/taproot.go

```diff
@@ -113,6 +115,49 @@ func NewPreimageFromBranch(branch txscript.TapBranch) *TapscriptPreimage {
 	}
 }
 
+// TapTreeToSibling constucts a taproot sibling hash from a Tapscript tree,
+// to be used with a TapCommitment tree root to derive a tapscript root. This
+// mimics the logic in the lnd/input package, and is needed here because the
+// Tapscript tree root hash is not returned when constructing a Tapscript
+// object.
+func NewPreimageFromTreePreimage(
+	p asset.TapscriptTreePreimage) (*TapscriptPreimage, error) {
+
+	var (
+		preimage *TapscriptPreimage
+		err      error
+	)
+
+	p.WhenLeft(func(tbp asset.TapBranchPreimage) {
+		preimage = &TapscriptPreimage{
+			SiblingPreimage: asset.EncodeTapBranchPreimage(tbp),
+			SiblingType:     BranchPreimage,
+		}
+	})
+	p.WhenRight(func(tl []txscript.TapLeaf) {
+		if len(tl) == 1 {
+			preimage = NewPreimageFromLeaf(tl[0])
+
+			// A single tapscript leaf must be verified to not be
+			// another Taproot Asset commitment before use.
+			err = preimage.VerifyNoCommitment()
+			return
+		}
+
+		tree := txscript.AssembleTaprootScriptTree(tl...)
+		rootChildren := txscript.NewTapBranch(
+			tree.RootNode.Left(), tree.RootNode.Right(),
+		)
+		preimage = NewPreimageFromBranch(rootChildren)
```

Comment:
Hmm, yeah that comment above is funky. The original logic was written assuming that the `SiblingPreimage` bytes here is actually 32-bytes, as it's already been hashed into something that goes into the actual tapscript tree. 

To remedy this, I think we need to partition from `NewPreimageFromLeaf` slightly to make a version that hashes one level. We should also consult the existing itest here: https://github.com/lightninglabs/taproot-assets/blob/eb3ed937f3c92e0d6a7315d914274ad0d4186cef/tapscript/send_test.go#L2169